### PR TITLE
project: delete highly undesirable unused strong deps

### PR DESCRIPTION
**Pull request by nsajko**
NB: ElectronDisplay is still a weak dep after this.

### CODE REVIEW

**Comment by brianguenter** [comment]
This caused an error in the documentation build:
```
Run julia --project=docs/ -e 'using Pkg; Pkg.develop(PackageSpec(path=pwd())); Pkg.instantiate()'
  Installing known registries into `~/.julia`
ERROR: Compat `ElectronDisplay` not listed in `deps` or `extras` section.
```

**Comment by brianguenter** [comment]
Once the documentation build error is fixed I'll approve the PR.

**Comment by nsajko** [comment]
The last commit bumps the oldest supported Julia version from v1.8 to v1.9, as v1.8 didn't support package extensions (which is basically unrelated to this PR).

In case you instead still want to support v1.8, it's possible to use https://github.com/cjdoris/PackageExtensionCompat.jl for compatibility.

File: .github/workflows/documentation.yml [hunk 1]
```diff
@@ -16,11 +16,11 @@ jobs:
       - uses: actions/checkout@v4
       - uses: julia-actions/setup-julia@v2
         with:
-          version: '1.8'
```
Comment (brianguenter) [inline .github/workflows/documentation.yml:19]: Here you've set the julia version to 1.10 but in the FastDifferentiation Project.toml you've set the Julia version to 1.9. Was there a reason to make the versions different?

Comment (nsajko) [reply .github/workflows/documentation.yml:19]: These two are independent.

The first version of Julia that supports package extensions is v1.9. Given that this package uses a package extension, it's necessary to either set the julia compat in your package's project to v1.9 or, as mentioned, to use [PackageExtensionCompat.jl](https://github.com/cjdoris/PackageExtensionCompat.jl).

The version in the documentation build is just the version of Julia used to build the documentation, there's no point in setting it to any other value than the latest supported version of Julia, which is v1.10.

**Review by brianguenter — APPROVED** [review]

**Pull request merged**